react/Gallery.js

```diff
@@ -1,34 +1,105 @@
 import React from 'react'
-import { PropTypes } from 'prop-types'
 import classNames from 'classnames'
+import { List, WindowScroller, AutoSizer, CellMeasurer, CellMeasurerCache } from 'react-virtualized'
+import { PropTypes } from 'prop-types'
+
+import { withRuntimeContext, NoSSR } from 'render'
 
 import { productShape } from './constants/propTypes'
 import GalleryItem from './components/GalleryItem'
 
 /**
  * Canonical gallery that displays a list of given products.
  */
-const Gallery = ({ products, summary, layoutMode }) => {
-  const classes = classNames('vtex-gallery pa3 bn', {
-    'vtex-gallery--two-columns': layoutMode === 'small',
+
+const Gallery = ({
+  products,
+  summary,
+  layoutMode,
+  runtime: { hints: { mobile } },
+  itemWidth }) => {
+  const TWO_ITEMS = 2
+
+  const cache = new CellMeasurerCache({
+    fixedWidth: true,
+    keyMapper: () => 1,
```

Comment:
this 1 is for itens too?

Reply:
Yes, but in this case I think that using named constants doesn't make too much sense. I'll add anyway to maintain consistency.